python/lsst/meas/base/forcedPhotCoadd.py

```diff
@@ -287,13 +287,17 @@ def runDataRef(self, dataRef, psfCache=None):
             exposure.getPsf().setCacheCapacity(psfCache)
         refCat = self.fetchReferences(dataRef, exposure)
 
-        measCat = self.measurement.generateMeasCat(exposure, refCat, refWcs,
-                                                   idFactory=self.makeIdFactory(dataRef))
+        try:
+            exposureId = coaddUtils.getExposureId(dataRef, coaddName=self.config.coaddName, includeBand=False)
+        except Exception as e:
+            self.log.warn("Setting exposureId to match Gen3 failed with: %s.  Falling back to "
+                          "Gen2 implementation.", e)
+            exposureId = int(dataRef.get(self.config.coaddName + "CoaddId"))
```

Comment:
Between this and the `except` block in `coadd_utils`, I think we'll get two warnings if any of this ever fails.  Best to remove one.